src/flatcollections-array/FlatArray/FlatArray.T/Functional/Functional.Chunk.cs

```diff
@@ -0,0 +1,40 @@
+namespace System;
+
+partial struct FlatArray<T>
+{
+    public FlatArray<FlatArray<T>> Chunk(int size)
+    {
+        if (size < 1)
+        {
+            throw InnerExceptionFactory.SizeOutsideBounds(nameof(size), size);
+        }
+
+        if (length == default)
+        {
+            return default;
+        }
+
+        var chunksLength = length / size;
+        if (length % size != default)
+        {
+            chunksLength++;
+        }
+
+        var chunks = new FlatArray<T>[chunksLength];
+        for (var i = 0; i < chunks.Length; i++)
+        {
+            var start = i * size;
+
+            var chunkLength = length - start;

```

Comment:
Would like to suggest renaming `chunkLength `to `effectiveSize`.

(To avoid possible ambiguity between `chunksLength` and `chunkLength`, and to correspond to `size` parameter name.)

```suggestion
            var effectiveSize = length - start;
```

Reply:
Fixed